modules/mapper-extras/src/test/java/org/elasticsearch/index/query/ScoreScriptUtilsTests.java

```diff
@@ -78,5 +91,15 @@ public void testSparseVectorFunctions() {
         CosineSimilaritySparse cosineSimilaritySparse = new CosineSimilaritySparse(queryVector);
         double result2 = cosineSimilaritySparse.cosineSimilaritySparse(dvs);
         assertEquals("cosineSimilaritySparse result is not equal to the expected value!", 0.78, result2, 0.1);
+
+        // test l1norm
+        L1NormSparse l1Norm = new L1NormSparse(queryVector);
+        double result3 = l1Norm.l1normSparse(dvs);
+        assertEquals("l1normSparse result is not equal to the expected value!", 485.18, result3, 0.1);
+
+        // test l2norm
+        L2NormSparse l2Norm = new L2NormSparse(queryVector);
+        double result4 = l2Norm.l2normSparse(dvs);
+        assertEquals("l2normSparse result is not equal to the expected value!", 301.36, result4, 0.1);
```

Comment:
These tests don't cover the cases mentioned above where the queryVector contains dimensions not present in the document vector and vice versa.